llvm/unittests/ADT/ArrayRefTest.cpp

```diff
@@ -104,6 +104,15 @@ TEST(ArrayRefTest, DropFront) {
   EXPECT_TRUE(AR1.drop_front(2).equals(AR2));
 }
 
+TEST(ArrayRefTest, ConsumeFront) {
+  static const int TheNumbers[] = {4, 8, 15, 16, 23, 42};
+  ArrayRef<int> AR1(TheNumbers);
+  ArrayRef<int> AR2(&TheNumbers[2], AR1.size() - 2);
+  EXPECT_EQ(AR1.consume_front(), 4);
+  EXPECT_EQ(AR1.consume_front(), 8);
```

Comment:
This should also check that the expected **reference** was returned

Reply:
Done.